resources/metadata/kubernetesdeploy.yaml

```
@@ -113,16 +113,26 @@ spec:
         secret: true
       - name: containerRegistrySecret
         description: Name of the container registry secret used for pulling containers from the registry.
+        longDescription: |-
+          Name of the container registry secret used for pulling containers from the registry.
+
+          **For `deployTool: helm/helm3`:**<br />
+          If `containerRegistryUser` and `containerRegistryPassword` are provided a secret is created on the fly and the information is passed to the helm template.<br />
+          Note: the secret will not be persisted in the Kubernetes cluster.
+
+          If neither `containerRegistryUser` nor `containerRegistryPassword` are provided it is expected that a secret with the configured name exists in the target Kubernetes cluster.<br />
```

```suggestion
          If neither `containerRegistryUser` nor `containerRegistryPassword` are provided, it is expected that a secret with the configured name exists in the target Kubernetes cluster.<br />
```

... not 100% sure about these suggestions.